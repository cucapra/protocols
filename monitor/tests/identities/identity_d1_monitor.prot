// ARGS: --wave identities/identity_d1.fst --instances identity_d1:Identity 
struct Identity {
  in a: u32,
  out s: u32,
}

fn slicing_ok<DUT: Identity>(in a: u32, out s: u32) {
  DUT.a := a;

  step();

  assert_eq(s[31:15], DUT.s[31:15]);

  fork();
  step();
}

