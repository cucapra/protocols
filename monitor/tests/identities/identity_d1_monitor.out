slicing_ok(1, 1)
